rust/src/data/ffi/mod.rs

```diff
@@ -603,7 +613,12 @@ pub extern "C" fn opendp_data__object_as_slice(obj: *const AnyObject) -> FfiResu
             }
         }
         TypeContents::GENERIC { name, args } => {
-            if name == &"HashMap" {
+            if name == &"Function" {
+                if args.len() != 2 { return err!(FFI, "Functions should have 2 type arguments").into(); }
+                let I = try_!(Type::of_id(&args[0]));
+                let O = try_!(Type::of_id(&args[1]));
```

Comment:
Different message and different names, but otherwise the same as the `else`. Consider factoring out?